src/coreclr/nativeaot/Runtime/unix/PalRedhawkUnix.cpp

```diff
@@ -968,22 +969,26 @@ static void ActivationHandler(int code, siginfo_t* siginfo, void* context)
         g_pHijackCallback((NATIVE_CONTEXT*)context, NULL);
         errno = savedErrNo;
     }
+
+    Thread* pThread = ThreadStore::GetCurrentThreadIfAvailable();
+    if (pThread)
```

Comment:
When can this be null?

Reply:
If someone else sends a signal to a random thread in our process, that has not seen managed code. 
Not sure if this can happen in practice, but it is cheap to check.

Reply:
I can imagine profilers sending signals to do stack probing, for example.

On MacOS we use SIGUSR1, so it is more likely to share the signal with somebody.

Reply:
Ok, makes sense.